---
title: "Differential Gene Expression"
author: "Stephen Pederson<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  fig.width = 10
)
```

```{r packages}
library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(cowplot)
library(ggfortify)
library(magrittr)
library(cqn)
library(ggrepel)
library(DT)
```

```{r options}
panderOptions("table.split.table", Inf)
panderOptions("big.mark", ",")
theme_set(theme_bw())
```

```{r config}
config <- here::here("config/config.yml") %>%
  read_yaml()
suffix <- paste0(config$tag)
sp <- config$ref$species %>%
  str_replace("(^[a-z])[a-z]*_([a-z]+)", "\\1\\2") %>%
  str_to_title()
```


# Setup

## Gene Annotations

```{r ah}
ah <- AnnotationHub() %>%
  subset(rdataclass == "EnsDb") %>%
  subset(str_detect(description, as.character(config$ref$release))) %>%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)
```

```{r ensDb}
ensDb <- ah[[1]]
genesGR <- read_rds(here::here("output/genesGR.rds"))
```

Gene annotations were again loaded from Ensembl Release `r ensemblVersion(ensDb)`.
The [previously defined](qc_aligned.html#Annotation_Setup) `GenomicRanges` object containing GC content and Gene Length was also loaded,
containing information for `r comma(length(genesGR))` genes.

## Sample Merging

```{r samples}
samples <- config$samples %>%
  here::here() %>%
  read_tsv() %>%
  mutate(
    Filename = paste0(sample, suffix)
  ) %>%
  dplyr::select(Filename, flowcell_id, treat, rep, desc, run) %>%
  mutate(name = str_extract(desc, "ZR[0-9].++")) %>%
  dplyr::select(-desc) %>%
  mutate_if(
    function(x){length(unique(x)) < length(x)},
    as.factor
  ) %>%
  mutate(
    treat = relevel(treat, ref = "Vehicle")
  )
```

### QC Issues

Given that `SLX-7489.D712_D503` was identified as being low quality in previous QC steps, this sample was removed for subsequent analysis.
An apparent sequencing repeat of this sample (Replicate 4, Vehicle-treated) was identified in Run2 and it was assumed that this was performed in response to the initial low quality run.

```{r}
samples <- samples %>%
  dplyr::filter(!grepl("SLX-7489.D712_D503", Filename))
```


```{r mergedSamples}
mergedSamples <- samples %>%
  group_by(name, rep, treat, run) %>%
  tally()
```

```{r tabSamples}
pander(
  mergedSamples,
  caption = "*Summary of sequencing total runs for all samples*"
)
```

This data set is slightly unique in that samples were sequenced across multiple lanes and flowcells.
Metadata was re-organised to enable summarising of counts across the multiple sequencing runs obtained for every sample.
The [initial PCA](qc_aligned.html#pca) showed that variability between replicates and flowcells was dwarfed by variability between samples.
As such, all counts were able to be merged to give a single set of counts for each individual sample.


```{r set_cols}
treat_cols <- hcl.colors(
  n = length(levels(mergedSamples$treat)), 
  palette = "Red-Blue", 
  rev = TRUE
  ) %>%
  setNames(levels(mergedSamples$treat)) 
rep_cols <- hcl.colors(
  n = length(levels(mergedSamples$rep)), 
  palette = "Spectral"
  ) %>%
  setNames(levels(mergedSamples$rep))
```


## Count Data

```{r counts}
counts <- here::here("data/aligned/counts/counts.out") %>%
  read_tsv(comment = "#") %>%
  dplyr::select(Geneid, ends_with("bam")) %>%
  rename_at(vars(ends_with("bam")), dirname) %>%
  rename_all(basename) %>%
  column_to_rownames("Geneid") %>%
  .[,!grepl("SLX-7489.D712_D503", colnames(.))]
```

```{r mergedCounts}
mergedCounts <- counts %>%
  rownames_to_column("gene_id") %>%
  pivot_longer(
    cols = -gene_id,
    names_to = "Filename",
    values_to = "counts"
  ) %>%
  left_join(samples) %>%
  group_by(
    gene_id, name, rep, treat
  ) %>%
  summarise(counts = sum(counts), .groups = "drop") %>%
  pivot_wider(
    id_cols = gene_id,
    values_from = counts,
    names_from = name
  ) %>%
  column_to_rownames("gene_id")
```

```{r genes2Keep}
minCPM <- 1.5
minSamples <- ncol(mergedCounts) / 2
genes2Keep <- mergedCounts %>%
  cpm() %>%
  is_greater_than(minCPM) %>%
  rowSums() %>%
  is_weakly_greater_than(minSamples) 
```

The criteria for a gene to be considered as detected, $>$ `r minCPM` counts per million (CPM) were required to observed in $\geq$ `r minSamples` samples.
This effectively ensured $\geq$ `r ceiling(min(colSums(mergedCounts))/1e6)` counts in at least `r minSamples` merged libraries.

Of the `r comma(nrow(counts))` genes contained in the annotation for this release, `r comma(sum(!genes2Keep))` genes  were removed as failing this criteria for detection, leaving `r comma(sum(genes2Keep))` genes for downstream analysis.

```{r plotDensities, fig.height=5, fig.cap="*Distributions of logCPM values on merged counts, A) before and B) after filtering of undetectable genes. Some differences between replicates were noted.*"}
a <- mergedCounts %>%
  cpm(log = TRUE) %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  as_tibble() %>%
  pivot_longer(
    cols = contains("ZR"),
    names_to = "name",
    values_to = "logCPM"
  ) %>%
  left_join(samples) %>%
  ggplot(aes(logCPM, stat(density), colour = rep, linetype = treat)) +
  geom_density() +
  scale_colour_manual(values= rep_cols) +
  labs(
    y = "Density",
    colour = "Replicate",
    linetype = "Treatment"
  )
b <- mergedCounts[genes2Keep,] %>%
  cpm(log = TRUE) %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  as_tibble() %>%
  pivot_longer(
    cols = contains("ZR"),
    names_to = "name",
    values_to = "logCPM"
  ) %>%
  left_join(samples) %>%
  ggplot(aes(logCPM, stat(density), colour = rep, linetype = treat)) +
  geom_density() +
  scale_colour_manual(values= rep_cols) +
  labs(
    y = "Density",
    colour = "Replicate",
    linetype = "Treatment"
  )
plot_grid(
  a + theme(legend.position = "none"), 
  b + theme(legend.position = "none"), 
  get_legend(a),
  labels = c("A", "B"),
  rel_widths = c(4, 4, 1),
  nrow = 1
)
```

```{r dge}
dge <- mergedCounts[genes2Keep,] %>%
  DGEList(
    samples = mergedSamples %>%
      as.data.frame() %>%
      set_rownames(.$name) %>%
      .[,colnames(.)],
    group = mergedSamples %>%
      as.data.frame() %>%
      set_rownames(.$name) %>%
      .[,colnames(.)] %>%
      pull(treat),
    genes = genesGR[rownames(.)]
  ) %>%
  calcNormFactors()
```


# QC

## Library Sizes

```{r plotLibSizes, fig.height=6, fig.cap = "*Library sizes after removal of undetectable genes. The common-use minimum library size of 10 million reads is shown as a dashed line.*"}
dge$samples %>%
  ggplot(aes(name, lib.size, fill = treat)) +
  geom_col() +
  geom_hline(yintercept = 1e7, linetype = 2) +
  facet_wrap(~treat, scales = "free_x") +
  scale_y_continuous(
    labels = comma, expand = expansion(c(0, 0.05))
  ) +
  scale_fill_manual(values= treat_cols) +
  labs(x = "Sample Name", y = "Library Size") +
  theme(legend.position = "none") 
```

Given that previous work had only assessed the libraries prior to merging, the library sizes were checked after merging replicate sequencing runs.
The median library size was found to be `r round(median(colSums(mergedCounts[genes2Keep,] ))/1e6, 1)` million reads which is just above the common-use minimum recommendation of 10 million reads/sample.

## PCA: Pre Normalisation

```{r pca}
pca <- dge %>%
  cpm(log = TRUE) %>%
  t() %>%
  prcomp() 
```


```{r plotPCA, fig.height=8, fig.width=8, fig.cap="*PCA on logCPM from merged counts. No clear structure attributable to DHT treatment is evident. PC1 appears to capture the variability between replicates whilst PC2 captures much of the response in Replicates 1 & 3. However, the DHT responses seen in the remaining samples are relatively minimal.*"}
a <- pca %>%
  autoplot(
    data = dge$samples, 
    x = 1, y = 3,
    colour = "rep", 
    shape = "treat", 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = "Replicate",
    shape = "Treatment"
  )
b <- pca %>%
  autoplot(
    data = dge$samples, 
    x = 2, y = 3,
    colour = "rep", 
    shape = "treat", 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = "Replicate",
    shape = "Treatment"
  )
c <- pca %>%
  autoplot(
    data = dge$samples, 
    x = 1, y = 2,
    colour = "rep", 
    shape = "treat", 
    size = 3
  ) +
  geom_segment(
    aes(
      x = Vehicle_PC1, xend = DHT_PC1, 
      y = Vehicle_PC2, yend = DHT_PC2,
      colour = rep
    ),
    data = . %>%
      dplyr::select(any_of(colnames(dge$samples)), PC1, PC2) %>% 
      pivot_longer(cols = starts_with("PC"), names_to = "PC", values_to = "value") %>% 
      pivot_wider(names_from = c("treat", "PC"), values_from = value, id_cols = c("rep")),
    alpha = 0.4,
    arrow = arrow()
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = "Replicate",
    shape = "Treatment"
  )
plot_grid(
  a + theme(legend.position = "none"), 
  b + theme(legend.position = "none"),
  c + theme(legend.position = "none"),
  get_legend(a),
  labels = LETTERS[1:3],
  nrow = 2
)
```


## Checks for GC and Length bias

Given the clear unexpected variability captured by PC1, the impact of GC content or length bias was assessed as a possible source.

Genes were divided in 10 approximately equal sized bins based on increasing length, and 10 approximately equal sized bins based on increasing GC content, with the final GC/Length bins being the combination 100 bins using both sets.
The contribution of each gene to PC1 and PC2 was assessed and a t-test performed on each bin.
This tests 

$$
H_0: \mu = 0 \text{ against } H_A: \mu \neq 0
$$

where $\mu$ represents the true contribution to PC1 of all genes in that bin.

If any bin makes a contribution to PC1 the mean will be clearly non-zero, whilst if there is no contribution the mean will be near zero.
In this way, the impact of gene length and GC content on variance within the dataset can be assessed.
As seen below, the contribution of GC content and gene length to PC1 is very clear, with a smaller contribution being evident across PC2.
As a result, Conditional Quantile Normalisation (`CQN`) is recommended in preference to the more common TMM normalisation.

```{r plotPCA_GC_Length, fig.height=6, fig.cap = "*Contribution of each GC/Length Bin to PC1 and PC2. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values. The number of genes in each bin is indicated by the circle size. The clear pattern across PC1 is unambiguous.*"}
dge$genes %>%
  dplyr::select(gene_id, ave_tx_len, gc_content) %>%
  mutate(
    GC = cut(
      x = gc_content,
      labels = seq_len(10),
      breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    Length = cut(
      x = ave_tx_len,
      labels = seq_len(10),
      breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin = paste(GC, Length, sep = "_"),
    PC1 = pca$rotation[gene_id, "PC1"],
    PC2 = pca$rotation[gene_id, "PC2"]
  ) %>%
  pivot_longer(
    cols = c("PC1", "PC2"),
    names_to = "PC",
    values_to = "value"
  ) %>%
  group_by(PC, GC, Length, bin) %>%
  summarise(
    Size = n(),
    mean = mean(value),
    sd = sd(value),
    t = t.test(value)$statistic,
    p = t.test(value)$p.value,
    adjP = p.adjust(p, method = "bonf")
  ) %>%
  ggplot(
    aes(Length, GC, colour = t, alpha = -log10(adjP), size = Size)
  ) +
  geom_point() +
  facet_wrap(~PC) +
  scale_colour_gradient2() +
  scale_size_continuous(range = c(1, 10)) +
  labs(alpha = expression(paste(-log[10], p))) +
  theme(
    panel.grid = element_blank(),
    legend.position = "bottom"
  ) 
```


Given these results, the considered options for analysis are:

- Use CQN normalisation and GLM-QL Fits
- Use limma/voom to take advantage of the ability to down-weight samples
- Exclude replicate 2, in combination with either of the above strategies

## Normalisation

```{r cqn}
cqNorm <- with(
  dge,
  cqn(
    counts= counts,
    x = genes$gc_content,
    lengths = genes$ave_tx_len
  )
)
dge$offset <- cqNorm$glm.offset
logCPM <- cqNorm$y + cqNorm$offset
```

```{r plotCQN, fig.height=6, fig.cap = "*Model fits used when applying CQN. replicates 1 & 3 are again quite divergent here. In particular, the long genes with low GC content appear to be where the primary differences are found, in keeping with the previous PCA analysis.*"}
a <- cqNorm$func1 %>%
  as.data.frame() %>%
  mutate(x = cqNorm$grid1) %>%
  pivot_longer(
    cols = any_of(colnames(dge)),
    names_to = "name",
    values_to = "QR fit"
  ) %>%
  left_join(dge$samples) %>%
  ggplot(
    aes(x, `QR fit`, colour = rep, group = name, linetype = treat)
  ) +
  geom_line() +
  labs(x = "GC content", colour = "Replicate", linetype = "Treatment")
b <- cqNorm$func2 %>%
  as.data.frame() %>%
  mutate(x = cqNorm$grid2) %>%
  pivot_longer(
    cols = any_of(colnames(dge)),
    names_to = "name",
    values_to = "QR fit"
  ) %>%
  left_join(dge$samples) %>%
  ggplot(
    aes(x, `QR fit`, colour = rep, group = name, linetype = treat)
  ) +
  geom_line() +
  labs(
    x = expression(paste(log[10], " Gene Length (kb)")),
    colour = "Replicate", linetype = "Treatment"
  )
plot_grid(
  a + theme(legend.position = "none"), 
  b + theme(legend.position = "none"),
  get_legend(a),
  nrow = 1,
  rel_widths = c(3, 3, 1)
)
```

```{r pcaPost}
pcaPost <- logCPM %>%
  t() %>%
  prcomp() 
```

```{r plotPcaPost, fig.height=8, fig.width=8, fig.cap="*PCA on logCPM after performing CQN. No specific improvement was noted as a result of CQN, unlike other cell-lines.*"}
a <- pcaPost %>%
  autoplot(
    data = dge$samples, 
    x = 1, y = 3,
    colour = "rep", 
    shape = "treat", 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = "Replicate",
    shape = "Treatment"
  )
b <- pcaPost %>%
  autoplot(
    data = dge$samples, 
    x = 2, y = 3,
    colour = "rep", 
    shape = "treat", 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = "Replicate",
    shape = "Treatment"
  )
c <- pcaPost %>%
  autoplot(
    data = dge$samples, 
    x = 1, y = 2,
    colour = "rep", 
    shape = "treat", 
    size = 3
  ) +
  geom_segment(
    aes(
      x = Vehicle_PC1, xend = DHT_PC1, 
      y = Vehicle_PC2, yend = DHT_PC2,
      colour = rep
    ),
    data = . %>%
      dplyr::select(any_of(colnames(dge$samples)), PC1, PC2) %>% 
      pivot_longer(cols = starts_with("PC"), names_to = "PC", values_to = "value") %>% 
      pivot_wider(names_from = c("treat", "PC"), values_from = value, id_cols = c("rep")),
    alpha = 0.4,
    arrow = arrow()
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = "Replicate",
    shape = "Treatment"
  )
plot_grid(
  a + theme(legend.position = "none"), 
  b + theme(legend.position = "none"),
  c + theme(legend.position = "none"),
  get_legend(a),
  labels = LETTERS[1:3],
  nrow = 2
)
```


```{r plotPCA_GC_post, fig.height=6, fig.cap = "*Contributions of gene length and GC content to PC1 shown both before and after CQN. The strong patterns seen before normalisation have clearly been reduced. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values. The number of genes in each bin is indicated by the circle size. The clear pattern across PC1 is unambiguous.*"}
dge$genes %>%
  dplyr::select(gene_id, ave_tx_len, gc_content) %>%
  mutate(
    GC = cut(
      x = gc_content,
      labels = seq_len(10),
      breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    Length = cut(
      x = ave_tx_len,
      labels = seq_len(10),
      breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin = paste(GC, Length, sep = "_"),
    `pre-CQN` = pca$rotation[gene_id, "PC1"],
    `post-CQN` = pcaPost$rotation[gene_id, "PC1"]
  ) %>%
  pivot_longer(
    cols = c("pre-CQN", "post-CQN"),
    names_to = "status",
    values_to = "value"
  ) %>%
  mutate(status = factor(status, levels = c("pre-CQN", "post-CQN"))) %>%
  group_by(status, GC, Length, bin) %>%
  summarise(
    Size = n(),
    mean = mean(value),
    sd = sd(value),
    t = t.test(value)$statistic,
    p = t.test(value)$p.value,
    FDR = p.adjust(p, method = "bonf")
  ) %>%
  ggplot(
    aes(Length, GC, colour = t, alpha = -log10(FDR), size = Size)
  ) +
  geom_point() +
  facet_wrap(~status) +
  scale_colour_gradient2() +
  scale_size_continuous(range = c(1, 10)) +
  theme(
    panel.grid = element_blank(),
    legend.position = "bottom"
  ) 
```

CQN clearly had a positive impact on the data, however, the exclusion of replicate 2 may still be an option.
Given the consistency seen with the post-CQN PCA, using a blocking approach will also be the preferred analytic approach.

# Differential Expression Analysis

## Main Results

A design matrix was formed giving each replicate it's own baseline and the common DHT response was then specified as the final column.
Following this, dispersions were estimated for the main `DGEList` object and the model fitted using the Quasi-likelihood GLM.

```{r fit}
X <- model.matrix(~0 + rep + treat, data = dge$samples) %>%
  set_colnames(str_remove(colnames(.), "treat"))
dge <- estimateDisp(dge, design = X, robust = TRUE)
fit <- glmQLFit(dge)
```


```{r topTable}
alpha <- 0.05
lambda <- log2(1.2)
topTable <- glmTreat(fit, coef = "DHT", lfc = lambda) %>%
  topTags(n = Inf) %>%
  .[["table"]] %>%
  as_tibble() %>%
  mutate(
    location = paste0(seqnames, ":", start, "-", end, ":", strand),
    rankingStat = -sign(logFC)*log10(PValue),
    signedRank = rank(rankingStat),
    DE = FDR < alpha
  ) %>%
  dplyr::select(
    gene_id, gene_name, logCPM, logFC, PValue, FDR, 
    location, gene_biotype, entrezid, ave_tx_len, gc_content, 
    rankingStat, signedRank, DE
  )
de <- dplyr::filter(topTable, DE)$gene_id
up <- dplyr::filter(topTable, DE, logFC > 0)$gene_id
down <- dplyr::filter(topTable, DE, logFC < 0)$gene_id
```


In order to detect genes which respond to DHT the Null Hypothesis ($H_0$) was specified to be a range around zero, instead of the conventional point-value.

$$
H_0: -\lambda \leq \mu \leq \lambda \\
\text{Vs.} \\
H_A: |\mu| > \lambda
$$

This characterises $H_0$ as being a range instead of being the singularity 0, with $\mu$ representing the true mean logFC.
The default value of $\lambda = \log_2 `r 2^lambda` = `r round(lambda, 3)`$ was chosen.
This removes any requirement for *post hoc* filtering based on logFC.
Using this approach, `r length(de)` genes were considered as DE to an FDR of `r alpha`.
Of these, `r length(up)` were up-regulated with the remaining `r length(down)` being down-regulated.

```{r plotMA, fig.height=6, fig.cap = "*MA plot with blue curve through the data showing the GAM fit. The curve for low-expressed genes indicated a potential bias remains in the data, whilst this is not evident for highly expressed genes. DE genes are highlighted in red.*"}
topTable %>%
  ggplot(aes(logCPM, logFC)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>%
      dplyr::filter(logFC > 1.5 | logFC < -1),
    show.legend = FALSE
  ) +
  geom_smooth(se = FALSE) +
  scale_colour_manual(
    values = c("grey50", "red")
  ) +
  theme(
    legend.position = "none"
  )
```


```{r plotVolcano, fig.height=8, fig.cap = glue("*Volcano plot showing significance against log fold-change. The common-use cutoff values for considering a gene as DE (|logFC| > 1) are shown in blue as a guide only as these were not used for this analysis.*")}
topTable %>%
  ggplot(aes(logFC, -log10(PValue))) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %>%
      dplyr::filter(PValue < 2e-6 | (DE & logFC < 0)),
    show.legend = FALSE
  ) +
  geom_vline(xintercept = c(-1,1), colour = "blue", linetype = 2) +
  labs(
    y = expression(paste(-log[10], "p"))
  ) +
  scale_colour_manual(
    values = c("grey50", "red")
  ) +
  theme(
    legend.position = "none"
  )
```


```{r printTopTable}
topTable %>%
  dplyr::filter(DE) %>%
  dplyr::select(-DE, -gc_content, -ave_tx_len, -contains("rank")) %>%
  mutate(
    across(c("logFC", "logCPM"), round, digits = 2)
  ) %>%
  datatable(
    rownames = FALSE, 
    caption = glue("All genes were considered as DE using the criteria of an FDR-adjusted p-value < {alpha}.")
  ) %>%
  formatSignif(
    columns = c("PValue", "FDR"),
    digits = 3
  )
```

### Top DE Genes

logCPM values for the most up and down-regulated genes were also inspected using simple boxplots.

```{r plotTopUp, fig.height=8, echo=FALSE, fig.cap="*Top ranked up-regulated genes*"}
topTable %>%
  dplyr::filter(DE & logFC > 0) %>%
  dplyr::slice(1:12) %>%
  dplyr::select(gene_id, gene_name) %>%
  mutate(gene_name = fct_inorder(gene_name)) %>%
  cbind(
    cpm(dge, log = TRUE)[.$gene_id,]
  ) %>%
  pivot_longer(
    any_of(colnames(dge)),
    names_to = "name",
    values_to = "logCPM"
  ) %>%
  left_join(dge$samples) %>%
  ggplot(
    aes(treat, logCPM, fill = treat)
  ) +
  geom_boxplot(size = 1/2) +
  facet_wrap(~gene_name) +
  scale_y_continuous(breaks = seq(-4, 8, by = 4)) +
  scale_fill_manual(values = treat_cols)
```

```{r plotTopDown, fig.height=4, fig.width=6, echo=FALSE, fig.cap = "*Top ranked down-regulated genes*"}
topTable %>%
  dplyr::filter(DE & logFC < 0) %>%
  dplyr::slice(1:12) %>%
  dplyr::select(gene_id, gene_name) %>%
  mutate(gene_name = fct_inorder(gene_name)) %>%
  cbind(
    cpm(dge, log = TRUE)[.$gene_id,]
  ) %>%
  pivot_longer(
    any_of(colnames(dge)),
    names_to = "name",
    values_to = "logCPM"
  ) %>%
  left_join(dge$samples) %>%
  ggplot(
    aes(treat, logCPM, fill = treat)
  ) +
  geom_boxplot(size = 1/2) +
  facet_wrap(~gene_name) +
  scale_y_continuous(breaks = seq(-4, 8, by = 4)) +
  coord_cartesian(ylim = c(0, 10)) +
  scale_fill_manual(values = treat_cols)
```

## Checks For Bias

Using the sign of logFC to indicate direction, and -$\log_{10} p$ as the ranking statistic, bias was also checked within the set of results, using GC content and gene length as potential sources of bias amongst the set of DE genes.
Given the extremely low number of genes considered as DE, this was only minimally informative

```{r plotBias, fig.height = 8, fig.cap = "*Genes ranked for signed differential expression shown against A) GC Content, and B) Gene Length. Ranks were assigned based on the ranking statistic R = -sign(logFC)*log~10~p, such that genes at the left are most down-regulated, whilst those at the right are the most up-regulated. Horizontal black lines indicate the overall average value, whilst blue curves indicate the localised GAM fit. A small positive bias was noted at both extremes for both GC content and Gene Length.*"}
a <- topTable %>%
  ggplot(aes(signedRank, gc_content)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = mean(topTable$gc_content)) +
  labs(
    x = "Rank", y = "% GC Content"
  ) +
  scale_colour_manual(values = c("grey50", "red")) + 
  theme(legend.position = "none")
b <- topTable %>%
    ggplot(aes(gc_content, stat(density))) +
    geom_density(colour = "grey50") +
    geom_vline(xintercept = mean(topTable$gc_content)) +
    coord_flip() + 
    theme(
      axis.text = element_blank(),
      axis.title = element_blank(),
      axis.ticks = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
c <- topTable %>%
  ggplot(aes(signedRank, ave_tx_len)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 10^mean(log10(topTable$ave_tx_len))) +
  labs(
    x = "Rank", y = "Gene Length (nt)"
  ) +
  scale_y_log10(label = comma) +
  scale_colour_manual(values = c("grey50", "red")) + 
  theme(legend.position = "none")
d <- topTable %>%
  ggplot(aes(ave_tx_len, stat(density))) +
  geom_density(colour = "grey50") +
  geom_vline(xintercept = 10^mean(log10(topTable$ave_tx_len))) +
  coord_flip() + 
  scale_x_log10() +
  theme(
    axis.text = element_blank(),
    axis.title = element_blank(),
    axis.ticks = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
plot_grid(
  plot_grid(ngsReports:::.emptyPlot(""), a, b, rel_widths = c(0.35, 9, 1), align = "h", nrow = 1),
  plot_grid(c, d, rel_widths = c(9, 1), align = "h"),
  nrow = 2,
  rel_heights = c(1, 1),
  labels = c("A", "B"),
  align = "v",
  axis = "l"
)
```

## Data Export

These above results were exported as the file `ZR75_DHT-RNASeq.tsv`.

```{r exportDE}
topTable %>%
  mutate(
    entrezid = vapply(entrezid, str_flatten, character(1), collapse = ";")
  ) %>%
  write_tsv(
    file = here::here("output/ZR75_DHT-RNASeq.tsv")
  )
```

In addition, the final `DGEList` was exported for easier integration with subsequent analyses

```{r exportDGE}
write_rds(
  dge,
  file = here::here("output/dge.rds")
)
```

